EventListener/VersionListener.php

```diff
@@ -53,16 +53,18 @@ public function onKernelRequest(GetResponseEvent $event)
         }
 
         $this->version = $this->versionResolver->resolve($request);
-        if (false === $this->version && null !== $this->defaultVersion) {
+        if (false === $this->version && null !== $this->defaultVersion && !$request->attributes->has('version')) {
             $this->version = $this->defaultVersion;
         }
```

Comment:
By the way, what was the reason for the `getVersion()` method which requires us to make the event listener stateful?